presto-pinot-toolkit/src/main/java/com/facebook/presto/pinot/query/PinotFilterExpressionConverter.java

```diff
@@ -171,9 +232,26 @@ public PinotExpression visitCall(CallExpression call, Function<VariableReference
                 return handleLogicalBinary(operatorType.getOperator(), call, context);
             }
         }
+        if (functionMetadata.getName().getFunctionName().equalsIgnoreCase("$literal$timestamp") ||
```

Comment:
Since `handleLogicalBinary` is called here (above), I wonder if we can move all of the special logic around handling of date/timestamps into a separate function that is called here without dirtying up `handleLogicalBinary`, as we have already done so for some class of functions on line 235

Reply:
This is the handling to allow queries like `millisSinceEpoch < 1391126400000` been handled correctly. typeManager.canCoerce will return false and directly fail this query.